src/realm/db.cpp

```diff
@@ -1077,22 +1077,22 @@ void DB::open(const std::string& path, bool no_create_file, const DBOptions opti
                 // make sure that all possible concurrent session participants
                 // use the same durability setting for the same Realm file.
                 if (Durability(info->durability) != options.durability)
-                    throw LogicError(LogicError::mixed_durability);
+                    throw RuntimeError(ErrorCodes::IncompatibleSession, "Durability not consistent");
 
                 // History type must be consistent across a session. An
                 // inconsistency is a logic error, as the user is required to
                 // make sure that all possible concurrent session participants
                 // use the same history type for the same Realm file.
                 if (info->history_type != openers_hist_type)
-                    throw LogicError(LogicError::mixed_history_type);
+                    throw RuntimeError(ErrorCodes::IncompatibleSession, "Historuy type not consistent");
```

Comment:
```suggestion
                    throw RuntimeError(ErrorCodes::IncompatibleSession, "History type not consistent");
```

Reply:
Wouldn't it be good for troubleshooting to add the differing history_types in the exception?

This is a common question across the board. Include as much info/context as possible to enable easier troubleshooting.